File: tests/test_second_module.py
"""Unit testing for second_module/second_module.py.

- Author: Jongkuk Lim
- Contact: limjk@jmarple.ai
"""

from scripts.second_module.second_module import demo_generate_world


def test_demo_generate_world() -> None:
    """Unit test for demo_generate_world function."""

    result1 = demo_generate_world(1)
    result2 = demo_generate_world(2)
    result3 = demo_generate_world(3)

    assert result1 == "world"
    assert result2 == "world" * 2
    assert result3 == "world" * 3


if __name__ == "__main__":
    test_demo_generate_world()

File: cpp/tests/test_second_module.cpp
/// @file
/// @author Jongkuk Lim <limjk@jmarple.ai>
/// @copyright 2021 J.Marple
/// @brief Unit testing for second_module

#include <gtest/gtest.h>

#include <string>

#include "trt_loader/trt_loader.hpp"

TEST(SecondModuleTest, GenerateWorldTest) {
  std::string test_result1 = second_module::demo_generate_world(1);
  std::string test_result2 = second_module::demo_generate_world(2);
  std::string test_result3 = second_module::demo_generate_world(3);

  EXPECT_EQ(test_result1, "world");
  EXPECT_EQ(test_result2, "worldworld");
  EXPECT_EQ(test_result3, "worldworldworld");
}

int main(int argc, char **argv) {
  testing::InitGoogleTest(&argc, argv);
  auto test_results = RUN_ALL_TESTS();

  return test_results;
}
